### PR TITLE
Migrated circle CI config to 2.0 

### PR DESCRIPTION
**Pull request by sbuliarca**
This version is close to the one in  [cookiecuter from v1](https://github.com/Financial-Times/cookiecutter-upp-golang/blob/master/%7B%7B%20cookiecutter.repo_name%20%7D%7D/circle.yml) but it doesn't use govendor for fetching dependencies, as vendoring is not introduced for this repo yet.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+version: 2
+jobs:
+  build:
+    docker:
+      - image: circleci/golang:1.7
```
Comment (nwrigh) [inline .circleci/config.yml]: This should be 1.8

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          name: "Compute coverage"
+          command: |
+            goveralls -coverprofile=$CIRCLE_ARTIFACTS/coverage.out -service=circle-ci -repotoken=$COVERALLS_TOKEN
+
+      - store_test_results:
+          path: /tmp/test-reports
+
+      - store_artifacts:
+          path: /tmp/artifacts
+          destination: build
```
Comment (nwrigh) [inline .circleci/config.yml:57]: We can also to a build of the docker image:

  docker_build:
    working_directory: /public-concordances-api
    docker:
      - image: docker:1.12.6-git
    steps:
      - checkout
      - setup_docker_engine
      - run:
          name: Build Dockerfile
          command: docker build .

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+          name: "Get build dependencies"
+          # Not using govendor for syncing deps as vendoring is not yet introduced for this repository
+          command: |
+            go get -u github.com/kardianos/govendor
+            # govendor sync
+            go get -t -d -v ./...
+      - run:
+          name: "Build project"
+          command: |
+            go build -v
```
Comment (nwrigh) [inline .circleci/config.yml]: > govendor build -v 

instead?

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      NEO4J_TEST_URL: "http://localhost:7474/db/data/"
+
+    steps:
+      - checkout
+      - run:
+          name: "Get build dependencies"
+          # Not using govendor for syncing deps as vendoring is not yet introduced for this repository
+          command: |
+            go get -u github.com/kardianos/govendor
+            # govendor sync
```
Comment (nwrigh) [inline .circleci/config.yml]: Why not govendor sync?

File: .circleci/config.yml (hunk 2 above)
Comment (sbuliarca) [reply .circleci/config.yml:57]: Cool, I added it here also.

File: .circleci/config.yml (hunk 1 above)
Comment (sbuliarca) [reply .circleci/config.yml]: Thanks, fixed.

File: .circleci/config.yml (hunk 4 above)
Comment (sbuliarca) [reply .circleci/config.yml]: I introduced vendoring also, so now we can use govendor.

File: .circleci/config.yml (hunk 3 above)
Comment (sbuliarca) [reply .circleci/config.yml]: Sure.

**Comment by sbuliarca** [comment]
Closing this one, as I realized vendoring was already introduced on master.

**Pull request closed without merging**